heroic-elasticsearch-utils/src/main/java/com/spotify/heroic/elasticsearch/AbstractElasticsearchMetadataBackend.java

```diff
@@ -103,27 +131,32 @@ public AbstractElasticsearchMetadataBackend(final AsyncFramework async, final St
         int duplicates = 0;
         final Set<T> results = new HashSet<>();
         final Function<SearchHit, T> converter;
-        final Function<String, Supplier<AsyncFuture<SearchResponse>>> scrollFactory;
+        final BiFunction<String, SearchHit, Supplier<AsyncFuture<SearchResponse>>> searchFactory;
+        final Boolean scrolling;
 
-        @java.beans.ConstructorProperties({ "async", "limit", "converter", "scrollFactory" })
-        public ScrollTransform(
+        public SearchTransform(
             final AsyncFramework async,
             final OptionalLimit limit,
             final Function<SearchHit, T> converter,
-            final Function<String, Supplier<AsyncFuture<SearchResponse>>> scrollFactory
+            final BiFunction<String, SearchHit, Supplier<AsyncFuture<SearchResponse>>>
```

Comment:
nit; this is a monster of a parameter... is this the way the formatter formatted it? 

is this more readable?
```
final BiFunction
    <String, SearchHit, Supplier<AsyncFuture<SearchResponse>>> searchFactory
```

Reply:
I don't like having generics separated like that - makes me double-take to realize its a `BiFunction<Thing>` rather than just a `BiFunction`. AFAIK there's no standard/best-practice style guide for this.

What about splitting in the middle of the generic sig, like:

```
BiFuncton<String, SearchHit,
          Supplier<AsyncFuture<SearchResponse>>> searchFactory
```

Reply:
that's more readable for me 👍